source/slang/slang.cpp

```diff
@@ -3196,14 +3256,21 @@ SLANG_API SlangResult spCompile(
         req->getSink()->diagnose(Slang::SourceLoc(), Slang::Diagnostics::compilationAborted);
     }
     req->mDiagnosticOutput = req->getSink()->outputBuffer.ProduceString();
-    return res;
+    
 #else
     // When debugging, we probably don't want to filter out any errors, since
     // we are probably trying to root-cause and *fix* those errors.
     {
-        return req->executeActions();
+        res = req->executeActions();
     }
 #endif
+
+    if (req->dumpRepro.getLength())
```

Comment:
Nice. The placement of this means that we can dump even on a failure that involved throwing an exception. It still doesn't protect against a "hard" compiler crash, but those are probably easier to debug in situ.

Reply:
I suppose we could add an option to capture with arbitrary exception too.